src/server/index.jsx

```diff
@@ -219,17 +219,21 @@ server
 
       data.path = urlPath;
 
-      res.status(status).send(
-        await renderDocument({
-          bbcOrigin,
-          data,
-          isAmp,
-          routes,
-          service,
-          url,
-          variant,
-        }),
-      );
+      const result = await renderDocument({
+        bbcOrigin,
+        data,
+        isAmp,
+        routes,
+        service,
+        url,
+        variant,
+      });
+
+      if (typeof result === 'object' && result.url) {
```

Comment:
Is result ever not an object? How can you tell we should redirect by checking the presence of an url?

Reply:
The result is usually a string. I modified the `renderDocument` function, to return the context, when a redirect occurs when rendering the app. This way, the express server . knows, and can perform a 301 redirect accordingly.

Reply:
I think any function that could return either a string or an object is confusing. I wonder if there's a better way 🤔

Reply:
Even with the name of `renderDocument` I'm expecting a document to be rendered and not potentially an object with an url that should be recognised as an url to redirect to.

Reply:
maybe a solution is to throw an error inside of `renderDocument` when it's a redirect? then in catch block below you could redirect. maybe not though because a redirect isn't really an error

Reply:
or we could change the result of renderDocument to always return objects. So for a redirect:

```js
{
  status: 301,
  redirectUrl: 'someUrl'
}
```

and for a rendered document

```js
{
  status: 200,
  html: 'theDocumentHTMLString'
}
```

Reply:
I like changing the result of renderDocument to always return an object.

Reply:
I think introducing `status` adds complexity we do not need. When returning the html, we do not know that it'll be a success result, as it could be a 404 or 500 result, yet we'd be returning `200`.

`{ html: '' }` and `{ redirectUrl: '' }` alone, should offer enough context to the nature of the result.